grafonnext/mkschema.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"cuelang.org/go/cue/ast"
+	"github.com/grafana/grafana/pkg/cuectx"
+	"github.com/grafana/grafana/pkg/plugins/pfs/corelist"
+	"github.com/grafana/grafana/pkg/registry/corekind"
+	"github.com/grafana/thema/encoding/jsonschema"
+)
+
+// this is a meaningless hack, we have to work out versioning
+var grafanaVersion = "latest"
+
+var rt = cuectx.GrafanaThemaRuntime()
+
+func main() {
+	// All core kinds first
+	base := filepath.Join("schemas", grafanaVersion, "core")
+	die(os.MkdirAll(base, 0777))
+
+	for _, kind := range corekind.NewBase(nil).AllStructured() {
```

Comment:
Probably the main thing that using the `kind` as represented in Go offers is all the [metadata/DSL](https://github.com/grafana/grafana/pull/56492/files#diff-c5cbe27448c5f0b5b871148235f2b50dd3089e7a3f133d8cf5500d3e0583f823) we'll be wrapping around the kind declarations. In a Go context, [these structs](https://github.com/grafana/grafana/pull/56492/files#diff-457e2380f878c9205fc7a3201f5af4693d1b5f962ef554b428898ebca4ccd810) will be kept in sync with what's in the cue framework definitions, and are easy to access. It will still probably be possible with generic CLI tooling - use `thema` to grab out lineages from known subpaths, and then `cue` to extract certain known metadata from other subpaths, emit it as JSON, etc. But i suspect that will get increasingly unwieldy compared to just using the system designed to make this all easy to access from Go, in an external repo, as we're doing here.

But for now, the only place where this really matters is probably in

Reply:
The main area of interest right now with this is probably maturity flags, btw - it seems wise to indicate something about the [maturity levels and corresponding guarantees](https://docs.google.com/document/d/1gPzSCQn-0Qg86PYpPcR-l_kTQ4Ba1wgnNLZFKPSe8yQ/edit#heading=h.ajykcvt65mfb) that come with them for each schema.

My stance on grok is that i'm not willing to put it out at all until we have clear comms in that regard. I understand how there are different considerations here, where there were never guarantees in the first place. So probably fine to not consider that part of MVP here.